lib/core/test/integration/Test/Integration/Scenario/CLI/Wallets.hs

```diff
@@ -424,20 +463,107 @@ spec = do
+                  , replicate passphraseMaxLength 'ź'
+                 , expect (ExitSuccess, "\n", cmdOk)
+                  )
+                ]
+        forM_ matrix $ \(title, ppOldRight, expectations) -> it title $ \ctx -> do
+            let name = "name"
+            let ppNew = replicate passphraseMaxLength 'ź'
+            wid <- emptyWalletWith' ctx
+                (name, T.pack ppOldRight, addressPoolGapMin)
+            (exitCode, out, err) <-
+                updateWalletPassphraseViaCLI @t ctx wid ppOldRight ppNew ppNew
+            expectations (exitCode, out, err)
+
+    describe "WALLETS_UPDATE_PASS_04 - Cannot update pass of wallets with false ids" $ do
+        forM_ falseWalletIds $ \(title, wid) -> it title $ \ctx -> do
+            let ppOld = "right secure passphrase"
+            let ppNew = "new secure passphrase"
+            (c, out, err) <-
+                updateWalletPassphraseViaCLI @t ctx wid ppOld ppNew ppNew
+            out `shouldBe` ""
+            c `shouldBe` ExitFailure 1
+            if (title == "40 chars hex") then
+                T.unpack  err `shouldContain`
+                    "I couldn't find a wallet with the given id:\
+                    \ 1111111111111111111111111111111111111111\n"
+            else
+                T.unpack  err `shouldContain`
+                    "wallet id should be an hex-encoded string of\
+                    \ 40 characters\n"
+
+    describe "WALLETS_UPDATE_PASS_05,06 - \
+        \Transaction after updating passphrase can only be made with new pass" $ do
+        let oldPass = "cardano-wallet"
+        let newPass = "cardano-wallet2"
+        let expectTxOK (ec, out, err) = do
+                ec `shouldBe` ExitSuccess
+                _ <- expectValidJSON (Proxy @(ApiTransaction t)) out
+                T.unpack err `shouldContain` cmdOk
+        let matrix =
+                [ ("Old passphrase -> fail", oldPass
+                  , expect (ExitFailure 1, mempty, errMsg403WrongPass)
+                  )
+                , ("New passphrase -> OK", newPass
+                  , expectTxOK
+                  )
+                ]
+
+        forM_ matrix $ \(title, pass, expectations) -> it title $ \ctx -> do
+            wSrc <- fixtureWallet ctx
+            wDest <- emptyWallet ctx
+            addr:_ <- listAddresses ctx wDest
+            let wid = T.unpack $ wSrc ^. walletId
+            (c, out, err) <-
+                updateWalletPassphraseViaCLI @t ctx wid oldPass newPass newPass
+            expect (ExitSuccess, "\n", cmdOk) (c, out, err)
+
+            let addrStr =
+                    encodeAddress (Proxy @t) (getApiT $ fst $ addr ^. #id)
+            let args = T.unpack <$>
+                    [ wSrc ^. walletId
+                    , "--payment", "1@" <> addrStr
+                    ]
+
+            (cTx, outTx, errTx) <- postTransactionViaCLI @t ctx pass args
+            expectations (cTx, outTx, errTx)
```

Comment:
The test seems to be lesser than its title. We check that we can make a transaction with the new passphrase, but we don't actually check that a transaction cannot be made using the old passphrase, right ?

Reply:
>  but we don't actually check that a transaction cannot be made using the old passphrase, right ?

We do!
```
        let matrix =
                [ ("Old passphrase -> fail", oldPass
                  , expect (ExitFailure 1, mempty, errMsg403WrongPass)
                  )
                , ("New passphrase -> OK", newPass
                  , expectTxOK
                  )
                ]
```

Reply:
Aaaaaah. Okay.